### PR TITLE
use more cesm style logging

### PR DESCRIPTION
**Pull request by jedwards4b**
This PR puts more of the MOM6 logging output into the place where cesm expects it to be.
It also partially addresses issue #227.   It will work best with https://github.com/ESCOMP/FMS_interface/pull/14
but there are no dependency requirements for the two to be merged together.

### CODE REVIEW

**Review by gustavo-marques — APPROVED** [review]
Thanks for doing this @jedwards4b

**Comment by jedwards4b** [comment]
I think that the tests are failing because github updated ubuntu-latest and that seems to have changed the netcdf install.

**Comment by marshallward** [comment]
See https://github.com/NOAA-GFDL/MOM6/pull/270 for a temporary fix to this problem.  (The tests fail under MPICH but work for OpenMPI.  We're still trying to understand why...)

**Comment by marshallward** [comment]
@jedwards4b I'd suggest the CI issue be fixed in dev/ncar rather than in this PR.  And I believe Gustavo has a plan to cherry-pick this commit into dev/ncar.  We're also going to find a way to do this smoothly into `main`.

**Comment by marshallward** [comment]
FWIW, dev/ncar will also fix the MacOS issue when the merge from `main` is completed.

(The trailing whitespace is on you, however :grinning: )

**Comment by jedwards4b** [comment]
I think that changing from actions/checkout@v2 to actions/checkout@v3 will solve most of these, but I will let you do that on dev/ncar

**Comment by marshallward** [comment]
Ah interesting thanks, I'll pass it along to @adcroft who set this up.

File: config_src/drivers/nuopc_cap/mom_cap.F90 [hunk 1]
```diff
@@ -5,16 +5,14 @@ module MOM_cap_mod
...
 use field_manager_mod,        only: field_manager_init, field_manager_end
-use fms_mod,                  only: fms_init, fms_end, open_namelist_file, check_nml_error
-use fms_mod,                  only: close_file, file_exist, uppercase
-use fms_io_mod,               only: fms_io_exit
-use mpp_domains_mod,          only: domain2d, mpp_get_compute_domain, mpp_get_compute_domains
+use mom_coms_infra,           only: MOM_infra_init, MOM_infra_end
+use mom_io_infra,             only: io_infra_end
+use mom_domain_infra,         only: get_domain_extent
+use MOM_io,                   only: stdout
+use mpp_domains_mod,          only: mpp_get_compute_domains
```
Comment (gustavo-marques) [inline config_src/drivers/nuopc_cap/mom_cap.F90:7]: Perhaps this can be replaced with
`use MOM_domains, only : get_domain_extent`

**Comment by alperaltuntas** [comment]
Is this PR fully ready now?

**Comment by gustavo-marques** [comment]
I think we still have direct calls to FMS that need to be replaced with the corresponding calls via MOM6 wrappers.

**Comment by jedwards4b** [comment]
Yes there are still some direct calls to FMS - I did what I could with limited knowledge.

File: config_src/drivers/nuopc_cap/mom_cap.F90 [hunk 2]
```diff
@@ -489,7 +486,33 @@ subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
...
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    call NUOPC_CompAttributeGet(gcomp, name="logfile", &
+         isPresent=isPresentLogfile, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    if (isPresentDiro .and. isPresentLogfile) then
+         call NUOPC_CompAttributeGet(gcomp, name="diro", value=diro, rc=rc)
+         if (ChkErr(rc,__LINE__,u_FILE_u)) return
+         call NUOPC_CompAttributeGet(gcomp, name="logfile", value=logfile, rc=rc)
+         if (ChkErr(rc,__LINE__,u_FILE_u)) return
+         open(newunit=stdout,file=trim(diro)//"/"//trim(logfile))
```
Comment (alperaltuntas) [inline config_src/drivers/nuopc_cap/mom_cap.F90:467]: @jedwards4b , here the `newunit` argument returns a negative number `-10`. Within FMS, unit numbers are used as array indices (see mpp_io_misc.inc), so when the unit is -10, i get the following error:

```
Fortran runtime error: Index '-10' of dimension 1 of array 'mpp_file' below lower bound of -1
```

You need to run a GNU debug test to reproduce the error.

Comment (jedwards4b) [reply config_src/drivers/nuopc_cap/mom_cap.F90:467]: Yes newunit returns a negative value - did you try with the latest FMS_Interface?   I said it wasn't required, but perhaps it is.

Comment (alperaltuntas) [reply config_src/drivers/nuopc_cap/mom_cap.F90:467]: Yes, I tried it with the latest FMS_interface and 2021.03.01

**Comment by jedwards4b** [comment]
I have a fix, it'll be a new fms_interface PR - coming soon.

**Comment by alperaltuntas** [comment]
@gustavo-marques I have removed more direct calls from FMS. The only remaining ones are subroutines from mpp_domains_mod. And these subroutines don't have `src/framework` or `config/src/infra/` interfaces. These subroutine calls appear to be added to accommodate multi tiles feature, which I believe are unnecessary.  We can either add interfaces or remove them altogether. But regardless, I think we should address them in a separate PR. So, I think this PR is ready now. Let us know if you have any other comments before I merge.

**Pull request merged**